### PR TITLE
[EN DateTimeV2] Added support for "To/Till date" (#2054)

### PR DESCRIPTION
**Pull request by LionbridgeCSII**
Added support for expressions like "to/till date" in DatePeriod (#2054).
Relevant test cases added to DateTimeModel.

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs [hunk 1]
```diff
@@ -190,6 +190,7 @@ public static class DateTimeDefinitions
       public static readonly string TimeNumberCombinedWithUnit = $@"\b(?<num>\d+(\.\d*)?){TimeUnitRegex}";
       public static readonly string[] BusinessHourSplitStrings = { @"business", @"hour" };
       public const string NowRegex = @"\b(?<now>(right\s+)?now|as soon as possible|asap|recently|previously)\b";
+      public static readonly string NowDateRegex = $@"\b({NowRegex}|^(date)$)\b";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs:193]: Perhaps rename this to "NowParseRegex"?
Also, wouldn't it be best to treat "^(date)$" as a named group to makes the parse behaviour explicit, as in onewordperiodregex? I suggest adding a comment on the YAML that this component wouldn't need to be translated .

**Review by tellarin — CHANGES_REQUESTED** [review]

File: .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs (hunk 1 above)
Comment (LionbridgeCSII) [reply .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs]: Renamed regex and added comment.
But I do not see the reason to use a named group here, "date" does not require a specific parse processing, it is handled as the other elements of NowRegex.

Comment (tellarin) [reply .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs]: Not a major issue. The point was to clarify for those localizing the regexes to other languages that this is English-specific and not exactly the same as "now".

Comment (tellarin) [reply .NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs]: I'm fine with merging as-is once conflicts are resolved.

**Review by tellarin — APPROVED** [review]
Only pending resolving conflicts.

**Comment by LionbridgeCSII** [comment]
> Only pending resolving conflicts.

solved

**Pull request merged**